sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -347,6 +347,7 @@ case class BroadcastPartitioning(mode: BroadcastMode) extends Partitioning {
   override val numPartitions: Int = 1
 
   override def satisfies0(required: Distribution): Boolean = required match {
+    case UnspecifiedDistribution => true
```

Comment:
@cloud-fan can you explain all these non-test changes in Pr description too? Seems like it's not test only

Reply:
This is the only one, as I don't want to special-case broadcast in the new `ValidateRequirements` like `EnsureRequirements`, but I can do that to avoid making this change (it doesn't have any real impact though)